docs/user/unattended/storage.md

```diff
@@ -0,0 +1,956 @@
+                }
+            ]
+        }
+    ]
+}
+```
+
+Other combinations can be used to specify how a device could be resized if possible. See the
+following examples with explanatory filesystem labels.
+
+```json
+"storage": {
+    "drives": [
+        {
+            "partitions": [
+                {
+                    "search": {
+                        "condition": { "property": "fsLabel", "value": "shrinkIfNeeded" }
+                    },
+                    "size": { "min": 0, "max": "current" }
+                },
+                {
+                    "search": {
+                        "condition": { "property": "fsLabel", "value": "resizeToFixedSize" }
+                    },
+                    "size": "15 GiB"
+                },
+                {
+                    "search": {
+                        "condition": { "property": "fsLabel", "value": "resizeByRange" }
+                    },
+                    "size": { "min": "10 GiB", "max": "50 GiB" }
+                },
+                {
+                    "search": {
+                        "condition": { "property": "fsLabel", "value": "growAsMuchAsPossible" }
+                    },
+                    "size": { "min": "current" }
+                },
+            ]
+        }
+    ]
+}
+```
+
+Of course, when the size limits are specified as a combination of "current" and a fixed value, the
+user must still make sure that the resulting min is not bigger than the resulting max.
+
+Both `deleteIfNeeded` and a size range can be combined to indicate that Agama should try to make
+space first by shrinking the partitions and deleting them only if shrinking is not enough.
+
+```json
+"storage": {
+    "drives": [
+        {
+            "partitions": [
+                {
+                    "search": {},
+                    "size": { "min": 0, "max": "current" },
+                    "deleteIfNeeded": true
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Generating Partitions as MD RAID members
+
+MD arrays can be configured to explicitly use a set of devices by adding their aliases to the
+`devices` property.
+
+```json
+"storage": {
+  "drives": [
+    {
+      "search": "/dev/sda",
+      "partitions": [
+        { "alias": "sda-40", "size": "40 GiB" }
+      ]
+    },
+    {
+      "search": "/dev/sdb",
+      "partitions": [
+        { "alias": "sdb-40", "size": "40 GiB" }
+      ]
+    }
+  ],
+  "mdRaids": [
+    {
+      "devices": [ "sda-40", "sdb-40" ],
+      "level": "raid0"
+    }
+  ]
+}
+```
+
+The partitions acting as members can be automatically generated by simply indicating the target
+disks that will hold the partitions. For that, the `devices` section must contain a `generate`
+entry.
+
+```json
+"storage": {
+  "drives": [
+    { "search": "/dev/sda", "alias": "sda" },
+    { "search": "/dev/sdb", "alias": "sdb" },
+  ],
+  "mdRaids": [
+    {
+      "devices": [
+        {
+          "generate": {
```

Comment:
I am not sure whether the concept of generation ("generate") deserves some introduction.

Reply:
Maybe... for a future pull request. :-)